File: code/HuaWeiNote/app/src/test/java/com/fan/hwnote/app/model/entity/NoteContentTest.kt
package com.fan.hwnote.app.model.entity

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

class NoteContentTest {

    @Test
    fun `toPlainText returns empty when blocks are empty`() {
        val content = NoteContent(blocks = emptyList(), handwriting = emptyList())
        assertEquals("", content.toPlainText())
    }

    @Test
    fun `toPlainText concatenates text blocks with newlines and skips images`() {
        val content = NoteContent(
            blocks = listOf(
                Block.TextBlock(id = "b1", text = "今天会议要点"),
                Block.ImageBlock(id = "b2", fileName = "x.jpg", width = 100, height = 100),
                Block.TextBlock(id = "b3", text = "结论：发版推迟"),
            ),
            handwriting = emptyList(),
        )
        assertEquals("今天会议要点\n结论：发版推迟", content.toPlainText())
    }

    @Test
    fun `toPlainText flattens checklist items and ignores empty blocks`() {
        val content = NoteContent(
            blocks = listOf(
                Block.TextBlock(id = "b1", text = "TODO"),
                Block.TextBlock(id = "b2", text = ""), // 空块跳过
                Block.ChecklistBlock(
                    id = "b3",
                    items = mutableListOf(
                        ChecklistItem(checked = true, text = "确认排期"),
                        ChecklistItem(checked = false, text = "联系李雷"),
                    ),
                ),
            ),
            handwriting = emptyList(),
        )
        assertEquals("TODO\n确认排期\n联系李雷", content.toPlainText())
    }
}

File: code/HuaweiNoteFlutter/test/models/note_content_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:hwnote/models/note_content.dart';
import 'package:hwnote/models/block.dart';
import 'package:hwnote/models/stroke.dart';

void main() {
  group('NoteContent', () {
    test('empty() creates empty content', () {
      final c = NoteContent.empty();
      expect(c.blocks, isEmpty);
      expect(c.handwriting, isEmpty);
    });

    test('toPlainText extracts text from TextBlock and ChecklistBlock', () {
      final c = NoteContent(
        blocks: [
          TextBlock(id: '1', text: 'Hello'),
          ImageBlock(id: '2', fileName: 'a.jpg', width: 1, height: 1),
          ChecklistBlock(id: '3', items: [
            ChecklistItem(checked: false, text: 'Buy milk'),
            ChecklistItem(checked: true, text: ''),
          ]),
          AudioBlock(id: '4', fileName: 'r.m4a', durationMs: 1000),
          TextBlock(id: '5', text: ''),
        ],
        handwriting: [],
      );
      expect(c.toPlainText(), 'Hello\nBuy milk');
    });

    test('toJson and fromJson roundtrip', () {
      final c = NoteContent(
        blocks: [TextBlock(id: '1', text: 'Hi')],
        handwriting: [
          Stroke(brush: BrushType.pen, color: '#000', width: 2, points: [
            StrokePoint(x: 1, y: 2, t: 0),
          ]),
        ],
      );
      final json = c.toJson();
      final restored = NoteContent.fromJson(json);
      expect(restored.blocks.length, 1);
      expect((restored.blocks[0] as TextBlock).text, 'Hi');
      expect(restored.handwriting.length, 1);
    });

    test('fromJson with invalid data returns empty', () {
      final c = NoteContent.fromJson('not json');
      expect(c.blocks, isEmpty);
      expect(c.handwriting, isEmpty);
    });
  });
}
